pkg/cmd/dedicated/listclusters/list_clusters.go

```diff
@@ -0,0 +1,176 @@
+	"github.com/redhat-developer/app-services-cli/pkg/shared/factory"
+	kafkamgmtclient "github.com/redhat-developer/app-services-sdk-go/kafkamgmt/apiv1/client"
+	"github.com/spf13/cobra"
+)
+
+type clusterRow struct {
+	ID            string `json:"id" header:"ID"`
+	Name          string `json:"name" header:"Name"`
+	Status        string `json:"status" header:"Status"`
+	CloudProvider string `json:"cloud_provider" header:"Cloud Provider"`
+	Region        string `json:"region" header:"Region"`
+}
+
+type options struct {
+	outputFormat            string
+	page                    int
+	limit                   int
+	search                  string
+	kfmClusterList          kafkamgmtclient.EnterpriseClusterList
+	selectedCluster         clustersmgmtv1.Cluster
+	clustermgmtClusterList  []clustersmgmtv1.Cluster
+	registeredClusters      []clusterRow
+	accessToken             string
+	clusterManagementApiUrl string
+
+	f *factory.Factory
+}
+
+const clusterReadyState = "ready"
+
+func NewListClusterCommand(f *factory.Factory) *cobra.Command {
+	opts := &options{
+		f: f,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "list",
+		Short:   f.Localizer.MustLocalize("dedicated.listClusters.cmd.shortDescription"),
+		Long:    f.Localizer.MustLocalize("dedicated.listClusters.cmd.longDescription"),
+		Example: f.Localizer.MustLocalize("dedicated.listClusters.cmd.example"),
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return runListClusters(opts, f)
+		},
+	}
+
+	flags := kafkaFlagutil.NewFlagSet(cmd, f.Localizer)
+	flags.IntVar(&opts.page, "page", int(cmdutil.ConvertPageValueToInt32(build.DefaultPageNumber)), f.Localizer.MustLocalize("dedicated.listClusters.flag.page"))
+	flags.IntVar(&opts.limit, "size", int(cmdutil.ConvertSizeValueToInt32(build.DefaultPageSize)), f.Localizer.MustLocalize("dedicated.listClusters.flag.limit"))
+
+	return cmd
+}
+
+func runListClusters(opts *options, f *factory.Factory) error {
+	err := listEnterpriseClusters(opts, f)
+	if err != nil {
+		return err
+	}
+	err = getPaginatedClusterList(opts)
+	if err != nil {
+		return err
+	}
+	err = mapRegisteredClusters(opts)
+	if err != nil {
+		return err
+	}
+	err = displayRegisteredClusters(opts)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func listEnterpriseClusters(opts *options, f *factory.Factory) error {
+	conn, err := f.Connection()
+	if err != nil {
+		return err
+	}
+	ctx := context.Background()
+	api := conn.API()
+	cl := api.KafkaMgmtEnterprise().GetEnterpriseOsdClusters(ctx)
+	clist, response, err := cl.Execute()
+	if err != nil {
+		return err
+	}
+	opts.kfmClusterList = clist
+	f.Logger.Info(response)
+	if len(opts.kfmClusterList.Items) == 0 {
+		return f.Localizer.MustLocalizeError("dedicated.listClusters.log.info.noClusters")
+	}
+	return nil
+}
+
+func getPaginatedClusterList(opts *options) error {
+	cl, err := clustermgmt.GetClusterList(opts.f, opts.accessToken, opts.clusterManagementApiUrl, opts.page, opts.limit)
```

Comment:
I'll suggest filtering by only the clusters that have been returned by `listEnterpriseClusters`.

i.e from `opts.kfmClusterList.Items`, you could map the ids and use them to make a call to OCM with a search parameter like:

```
ocm get /api/clusters_mgmt/v1/clusters --parameter search="id = 'id1' or id = 'id2' or .... id='idN'"
```
This will reduce the need for https://github.com/redhat-developer/app-services-cli/blob/bf1a206b76c6be17aa95ed2e793f6235148f83ef/pkg/cmd/dedicated/listclusters/list_clusters.go#L116 (even though this can still be removed with what you current have) or https://github.com/redhat-developer/app-services-cli/blob/bf1a206b76c6be17aa95ed2e793f6235148f83ef/pkg/cmd/dedicated/listclusters/list_clusters.go#L145

Reply:
When you use the search suggestion - you'll then make one call to ocm, the search parameter and the `opts.limit` to control the bucket size will then be set to `len(opts.kfmClusterList.Items)` the length of the items returned by registered enterprise clusters.

Reply:
Well that is a much better approach, will sort this 👍